kore/src/Kore/Equation/Application.hs

```diff
@@ -176,6 +216,28 @@ attemptEquation sideCondition termLike equation =
         debugAttemptEquationResult equation result
         return result
 
+    simplifyArgumentWithResult
+        :: HasCallStack
+        => MonadTrans t
+        => Predicate (Target variable)
+        -> Maybe (Predicate (Target variable))
+        -> MatchResult (Target variable)
+        -> t simplifier [MatchResult (Target variable)]
```

Comment:
Isn't `simplifyArgumentWithResult` called only in a particular instance of `t`? In that case, we should not make `t` a variable here, because it may prevent specialization. It would be enough just to remove the signature.